sdcm/nemesis.py

```diff
@@ -2429,26 +2429,6 @@ def disrupt_memory_stress(self):
             "stress-ng --vm-bytes $(awk '/MemTotal/{printf \"%d\\n\", $2 * 0.9;}' < /proc/meminfo)k --vm-keep -m 1 -t 100")
 
 
-class NotSpotNemesis(Nemesis):
-    def set_target_node(self):
-
-        if isinstance(self.cluster, ScyllaAWSCluster):
-            non_seed_nodes = [node for node in self.cluster.nodes
-                              if not node.is_seed and
-                              not node.is_spot and
-                              not node.running_nemesis]
-            if non_seed_nodes:
-                self.target_node = random.choice(non_seed_nodes)
-
-        else:
-            super(NotSpotNemesis, self).set_target_node()
-
-        self.log.info('Current Target: %s', self.target_node)
-
-    def disrupt(self):
-        raise NotImplementedError()
-
-
```

Comment:
First, this change is not correct and not requested.
The idea with this class is to not run this nemesis when the instances are Spot because we cannot stop spot instance.

Second, even if you think this change is required, the commit message should have been much clearer and explanatory on why the change is required.

Reply:
When I tried to run the longevity only with RestartThenRepairNodeMonkey, it failed due to this class. When I asked @fruch about it, he said that this nemesis is unnecessary and that I should remove it in a second commit

Reply:
Yes this `NotSpotNemesis` should go, but it's not the goal of this PR, it's a bit backwards the goal is addressing scylladb/scylla#7588

Reply:
> When I tried to run the longevity only with RestartThenRepairNodeMonkey, it failed due to this class. When I asked @fruch about it, he said that this nemesis is unnecessary and that I should remove it in a second commit

Maybe you tried with a spot instance?

Reply:
Nope, on demand